scripts/python/packages/parthenon_tools/parthenon_tools/phdf_diff.py

```diff
@@ -452,75 +470,80 @@ def compare(
         ]:
             continue
 
-        # initialize info values
-        same = True
-        errMax = -1.0
-        maxPos = [0, 0, 0]
-
         # Get values from file
-        val0 = f0.Get(var)
-        val1 = f1.Get(var)
-        isVec = np.prod(val0.shape) != f0.TotalCells
-        for idx, v in enumerate(val0):
-            if f0.isGhost[idx % f0.CellsPerBlock]:
-                # only consider real cells
-                continue
-            [ib, bidx, iz, iy, ix] = f0.ToLocation(idx)
-
-            # find location in other file
-            [idx1, ib1, bidx1, iz1, iy1, ix1] = otherLocations[idx]
-
-            # compute error
-            errVal = np.abs(v - val1[idx1])
-            errMag = np.linalg.norm(errVal)
-
-            # Note that we use norm / magnitude to compute error
-            if errMag > errMax:
-                errMax = errMag
-                errMaxPos = [f0.x[ib, ix], f0.y[ib, iy], f0.z[ib, iz]]
-
-            if np.linalg.norm(errVal) > tol:
-                same = False
-                no_diffs = False
-                if brief or quiet:
-                    breakOut = True
-                    break
-
-                if isVec:
-                    s = "["
-                    for xd in errVal:
-                        if xd == 0.0:
-                            s += " 0.0"
-                        else:
-                            s += " %10.4g" % xd
-                    s += "]"
-                else:
-                    s = "%10.4g" % errVal
-                if all:
-                    print(
-                        "  %20s: %6d: diff=" % (var, idx),
-                        s.strip(),
-                        "at:f0:%d:%.4f,%.4f,%.4f"
-                        % (idx, f0.x[ib, ix], f0.y[ib, iy], f0.z[ib, iz]),
-                        ":f1:%d:%.4f,%.4f,%.4f"
-                        % (idx1, f1.x[ib1, ix1], f1.y[ib1, iy1], f1.z[ib1, iz1]),
-                    )
-        if breakOut:
-            if not quiet:
-                print("")
-            print("Files %s and %s are different" % (f0.file, f1.file))
-            if not quiet:
-                print("")
-            break
-        if not quiet:
-            if same:
-                print("  %20s: no diffs" % var)
+        val0 = f0.Get(var, flatten=False)
+        val1 = f1.Get(var, flatten=False)
+
+        is_vec = np.prod(val0.shape) != f0.TotalCells
+
+        # Determine arrangement of mesh blocks of f1 in terms of ordering in f0
+        otherBlockIdx = list(f0.findBlockIdxInOther(f1, i) for i in range(f0.NumBlocks))
+
+        # Rearrange val1 to match ordering of meshblocks in val0
+        val1 = val1[otherBlockIdx]
+
+        # compute error at every point
+        if relative:
+            err_val = np.abs((val0 - val1) / val0)
+            # Set error values where val0==0 to 0
+            # Numpy masked arrays would be more robust here, but they are very slow
+            err_val[val0 == 0] = 0
```

Comment:
For symmetry (so that comparison results are the same if `f0` and `f1` are swapped), I'd propose a small change:
```suggestion
            denom = 0.5 * (np.abs(val0) + np.abs(val1))
            err_val = np.abs(val0 - val1) / denom
            # Set error values where denom==0 to 0
            # Numpy masked arrays would be more robust here, but they are very slow
            err_val[denom == 0] = 0
```

Reply:
I was going to suggest the exact same thing, @jlippuner great minds think alike. ;)

Reply:
Thirded.